### PR TITLE
Linear model definition

### PR DESCRIPTION
**Pull request by nieznanysprawiciel**
Specification created because of problem:
https://github.com/golemfactory/yagna/issues/3118

### CODE REVIEW

File: standards/3-commercial/com/pricing/linear-model.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+```
+price = usage_1 x coeffs_1 + ... + usage_N x coeffs_N + 1 x coeff_fixed.
+```
+
+where `n` denotes an index of a particular measure agreed between the provider and the requestor to be used for this agreement and `usage` is the actual resource usage per counter.
+
+## Usage Counters
+
+Usage `counters` are measures of resource consumption during the execution of the task or service. Each ExeUnit implements a specific subset of usage counters and reports their values to the Yagna daemon. It is the Provider's choice to decide which counters he uses to measure the resource consumption and finally calculate the price. Provider provides this information in the offer as the `golem.usage.vector`.
```
Comment (jalas167) [inline standards/3-commercial/com/pricing/linear-model.md:34]:
```suggestion
Usage `counters` are measures of resource consumption during the execution of the task or service. Each ExeUnit implements a specific subset of usage counters and reports their values to the Yagna daemon. It is the Provider's choice to decide which counters he uses to measure the resource consumption and finally calculate the price. Provider provides this information in the offer as the `golem.com.usage.vector`.
```

File: standards/3-commercial/com/pricing/linear-model.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+   "golem.usage.duration_sec"]`
+ which indicates that the price will be a function of both the time of CPU and the environment run. The actual price value will depend on the actual coeffs values.
+
+The order of the counters in the `golem.com.usage.vector` is important. The same order is used for the price coefficient array, so the coefficient on position n (`coeff_n`) corresponds to the usage counter on position n (`counter_n`). The provider will report the consumption also using this order.
+
+Usage counters are reported from the moment ExeUnit process starts.
+
+## Price Coefficients
+
+Price coefficients `golem.com.pricing.model.linear.coeffs` on the other side, contain prices of specific counters declared in the `usage` vector. Notice that the `coeffs` array has one element more than the `counters` vector, which always represents a fixed cost for starting an Activity. These values are decided by the provider and are subject to the agreement.
```
Comment (jalas167) [inline standards/3-commercial/com/pricing/linear-model.md:46]:
```suggestion
Price coefficients `golem.com.pricing.model.linear.coeffs` on the other side, contain prices of specific counters declared in the `usage` vector. Notice that the `coeffs` array has one element more than the `counters` vector, which always represents a fixed cost for starting an activity. These values are decided by the provider and are subject to the agreement.
```

File: standards/3-commercial/com/pricing/linear-model.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+## Usage Counters
+
+Usage `counters` are measures of resource consumption during the execution of the task or service. Each ExeUnit implements a specific subset of usage counters and reports their values to the Yagna daemon. It is the Provider's choice to decide which counters he uses to measure the resource consumption and finally calculate the price. Provider provides this information in the offer as the `golem.usage.vector`.
+
+A `golem.com.usage.vector` may be set as i.e. `["golem.usage.cpu_sec",
+   "golem.usage.duration_sec"]`
+ which indicates that the price will be a function of both the time of CPU and the environment run. The actual price value will depend on the actual coeffs values.
+
+The order of the counters in the `golem.com.usage.vector` is important. The same order is used for the price coefficient array, so the coefficient on position n (`coeff_n`) corresponds to the usage counter on position n (`counter_n`). The provider will report the consumption also using this order.
```
Comment (jalas167) [inline standards/3-commercial/com/pricing/linear-model.md:40]:
```suggestion
The order of the counters in the `golem.com.usage.vector` is important. The same order is used for the price coefficient array, so the coefficient on position n (`coeff_n`) corresponds to the usage counter on position n (`counter_n`). The provider will report the consumption using this order.
```

**Review by jalas167 — COMMENTED** [review]
one missing 'com' in the param name, the rest are formatting

File: standards/3-commercial/com/pricing/linear-model.md (hunk 2 above)
Comment (nieznanysprawiciel) [reply standards/3-commercial/com/pricing/linear-model.md:46]: In most of our docs we write yagna artifacts with capital letter

File: standards/3-commercial/com/pricing/linear-model.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+Usage counters are reported from the moment ExeUnit process starts.
+
+## Price Coefficients
+
+Price coefficients `golem.com.pricing.model.linear.coeffs` on the other side, contain prices of specific counters declared in the `usage` vector. Notice that the `coeffs` array has one element more than the `counters` vector, which always represents a fixed cost for starting an Activity. These values are decided by the provider and are subject to the agreement.
+
+## Computing cost
+
+Resource usage vector `usage` is reported by the ExeUnit - a part of the code responsible for running activities on the provider node. The ExeUnit is initialized with an Agreement containing the counters vector. During the Activity execution, it reports the vector of usage per counters according to the order found in the counters vector.
```
[the remark this answers is not in the document]
Comment (nieznanysprawiciel) [reply standards/3-commercial/com/pricing/linear-model.md:50]: In most of our docs we write yagna artifacts with capital letter

Comment (nieznanysprawiciel) [inline standards/3-commercial/com/pricing/linear-model.md:50]:
```suggestion
Resource usage vector `usage` is reported by the ExeUnit - a part of the code responsible for running activities on the provider node. The ExeUnit is initialized with an Agreement containing the counters vector. During the Activity execution, it reports the vector of usage per counters according to the order found in the `counters` vector.
```

**Review by jalas167 — APPROVED** [review]

**Pull request merged**